src/actions/arcActions.ts

```diff
@@ -572,11 +572,23 @@ export function stakeProposal(daoAvatarAddress: string, proposalId: string, vote
       const votingMachineAddress = schemeParams[2]; // 2 is the index of the votingMachine address for the ContributionReward scheme
       const votingMachineInstance = await Arc.GenesisProtocol.at(votingMachineAddress);
 
-      //const stakeTransaction = await votingMachineInstance.stake({ proposalId : proposalId, vote : vote, amount : web3.toWei(1, "ether")});
-      //console.log("Stake tr = ", stakeTransaction);
+      const votingMachineParamHash = await daoInstance.controller.getSchemeParameters(votingMachineInstance.contract.address, daoInstance.avatar.address);
+      const votingMachineParam = await votingMachineInstance.contract.parameters(votingMachineParamHash);
+      const minimumStakingFee = votingMachineParam[6]; // 6 is the index of minimumStakingFee in the Parameters struct.
 
-      const yesStakes = 0;//await votingMachineInstance.getVoteStake({ proposalId: proposalId, vote: VoteOptions.Yes });
-      const noStakes = 0; //await votingMachineInstance.getVoteStake({ proposalId: proposalId, vote: VoteOptions.No });
+      const input = parseInt(prompt(`How much ether would you like to stake? (min = ${minimumStakingFee})`, '1'));
+      const amount = web3.toWei(input,'ether');
+      if(amount < minimumStakingFee) throw new Error(`Staked less than the minimum: ${minimumStakingFee}!`);
+
+      const StandardToken = Arc.Utils.requireContract('StandardToken');
+      const stakingToken = await StandardToken.at(await votingMachineInstance.contract.stakingToken())
+      await stakingToken.approve(votingMachineInstance.address,amount);
```

Comment:
anyone can approve? or do they have to have reputation in the DAO?

Reply:
approve means :  approve the passed address to spend the specified amount of tokens on behalf of msg.sender.

In this case ,  the staker  approve the votingMachineInstance to spend 'amount' of tokens in its behalf.
It has nothing to do with reputation in the DAO.

Reply:
@dev-matan-tsuberi the approval and stake suppose to be from the staker account.

> e.g await stakingToken.approve(votingMachineInstance.address,amount,{from: staker});

what i am missing here ?

Reply:
@orenyodfat 
you can see [here](https://github.com/daostack/arc.js/blob/3763ae8c1274d370c0f0576484fa9fe7c6005c5b/lib/utils.ts#L35) and [here](https://github.com/daostack/arc.js/blob/3763ae8c1274d370c0f0576484fa9fe7c6005c5b/lib/utils.ts#L163) that arc.js changes the `web3.eth.defaultAccount` and `from` argument to be `accounts[0]` automatically.

Reply:
@tibetsprague 
This approval has nothing to do with DAOstack and reputation and everything to do with the `ERC20` standard which requires the owner of tokens to approve contracts (or humans) to spend tokens on his behalf (via the `transferFrom(..)` function).

Because `GenesisProtocol` wants to perform a `transferFrom` when you call `stake(..)` it needs an approval to do so from the owner.

Reply:
@dev-matan-tsuberi so the staker is  accounts[0] ?

Reply:
@orenyodfat yes